From b54faf431153535ddfa047f3ed45780b7cac5781 Mon Sep 17 00:00:00 2001
From: Aliaksey Kandratsenka <alk@tut.by>
Date: Sat, 13 Jun 2015 21:28:28 -0700
Subject: [PATCH 2/5] issue-693: convert sys_futex to it's 6-arg form

Because sys_futex actually takes 6 args in more recent kernels (even
though last two args are unused for FUTEX_{WAKE,WAIT}.

This is patch contributed by user spotrh.
---
 src/third_party/gperftools-2.2/src/base/linux_syscall_support.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git src/third_party/gperftools-2.2/src/base/linux_syscall_support.h src/third_party/gperftools-2.2/src/base/linux_syscall_support.h
index 9b6c35d..c8f8f59 100644
--- src/third_party/gperftools-2.2/src/base/linux_syscall_support.h
+++ src/third_party/gperftools-2.2/src/base/linux_syscall_support.h
@@ -2094,9 +2094,11 @@ struct kernel_stat {
                        int,            c, long,   a)
   LSS_INLINE _syscall2(int,     fstat,           int,         f,
                       struct kernel_stat*,   b)
-  LSS_INLINE _syscall4(int,     futex,           int*,        a,
+  LSS_INLINE _syscall6(int,     futex,           int*,        a,
                        int,            o, int,    v,
-                      struct kernel_timespec*, t)
+                      struct kernel_timespec*, t,
+                       int*, a2,
+                       int, v3)
 #ifdef __NR_getdents64
     LSS_INLINE _syscall3(int,     getdents64,      int,         f,
                          struct kernel_dirent64*, d, int,    c)
-- 
2.6.4

